homeassistant/components/elmax/common.py

```diff
@@ -92,54 +130,41 @@ def http_client(self):
         """Return the current http client being used by this instance."""
         return self._client
 
+    @http_client.setter
+    def http_client(self, client: GenericElmax):
+        """Set the client library instance for Elmax API."""
+        self._client = client
+
     async def _async_update_data(self):
         try:
-            async with asyncio.timeout(DEFAULT_TIMEOUT):
-                # Retrieve the panel online status first
-                panels = await self._client.list_control_panels()
-                panel = next(
-                    (panel for panel in panels if panel.hash == self._panel_id), None
-                )
-
-                # If the panel is no more available within the given. Raise config error as the user must
-                # reconfigure it in order to  make it work again
-                if not panel:
-                    raise ConfigEntryAuthFailed(
-                        f"Panel ID {self._panel_id} is no more linked to this user"
-                        " account"
-                    )
```

Comment:
Unfortunately, this can happen. 

The Elmax installer (or the user, if he has access to the admin console) can disassociate a panel from his account. 
If this happens while the user had the HA system down, we want to check this specific case and make the config entry fail with the AuthFailure message.

Reply:
Yes, I understand why it's checked. My question is why, with the changes in this PR, this check only happens when setting up the config entry instead of every time the data coordinator updates?

If cloud is used, and the panel is disassociated from the account when Home Assistant is already running, what happens now?

Reply:
If that happens, the `_async_update_data()` method of `ElmaxCoordinator` will hit a `ElmaxBadPinError` and trigger the `ConfigEntryAuthFailed`. That will make sure the user re-initiates the configuration with the new panel (if any).